src/app/core/connection.ts

```diff
@@ -153,6 +153,19 @@ export default class Connection {
      */
     updateParameter(key: string, value: any): void {
         this._params.updateParameter(key, value);
+
+        console.log("Key: ", key, " Val: ", value);
+
+        if (key != "segments") {
+            for (const i in this._featureIDs) {
+                const featureidtochange = this._featureIDs[i];
+
+                // Get the feature id and modify it
+                const feature = ConnectionUtils.getFeatureFromID(featureidtochange);
+                feature.updateParameter(key, value);
+            }
+        }
+
```

Comment:
We can leave this for now but we need to port the connection class to use the waypoints system and autocompute all the segment-related steps (and modify the drawing code for connections). We can schedule this after @msisk23  is done with the curved connections update